sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/NullPointerDereferenceCSharp8.cs

```diff
@@ -0,0 +1,148 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections;
+
+namespace Tests.Diagnostics.CSharp8
+{
+    public class NullCoalescenceAssignment
+    {
+        public void NullCoalescenceOperator_Null()
+        {
+            string name = null;
+            name = name ?? "Laxmi";
+            name.ToString();
+        }
+
+        public void NullCoalescenceOperator_Null_Noncompliant1()
+        {
+            string name = null;
+            name = name ?? null;
+            name.ToString(); // Noncompliant {{'name' is null on at least one execution path.}}
+        }
+
+        public void NullCoalescenceOperator_Null_Noncompliant2()
+        {
+            string name = null;
+            name = name ?? name.ToString(); // Noncompliant
+        }
+
+        public void NullCoalescenceAssignment_Null()
+        {
+            string name = null;
+            name ??= name.ToString(); // Noncompliant
+        }
+
+        public void NullCoalescenceAssignment_NotNull_Noncompliant()
+        {
+            string name1 = null;
+            string name2 = null;
+            name1 ??= name2.ToString(); // Noncompliant {{'name2' is null on at least one execution path.}}
+        }
+
+        public void NullCoalescenceAssignment_NotNull()
+        {
+            string name1 = "";
+            string name2 = null;
+            name1 ??= name2.ToString(); //  Ok - name1 is not null
+        }
+    }
+
+    public class NullablePrimitiveType
+    {
+        public void NullablePrimitiveType_FN()
+        {
+            int? i1 = null;
+            i1 = i1 ?? i1.GetHashCode(); // FN - nullable primitive type not supported
+        }
+    }
+
+    public class SwitchExpressions
+    {
+        public void Nullable_In_Arm_Noncompliant(string s)
+        {
+            var result = s switch
+            {
+                null => s.ToString(), // Noncompliant
+                _ => s.ToString()
+            };
+        }
+
+        public void AlwaysNull_Noncompliant(int val)
+        {
+            string result = val switch
+            {
+                1 => null,
+                2 => null,
+                _ => null
+            };
+            result.ToString(); // Noncompliant
+        }
+
+        public void MaybeNull_Noncompliant(int val)
+        {
+            string result = val switch
+            {
+                1 => "1",
+                2 => "Not 1",
+                _ => null
+            };
+            result.ToString(); // Noncompliant
+        }
+
+        public void AlwaysNonNull(int val)
+        {
+            string result = val switch
+            {
+                1 => "1",
+                2 => "2",
+                _ => "Neither 1 or 2"
+            };
+            result.ToString();
+        }
+
+        public void Nullable_In_Arm(string? s)
```

Comment:
Is that `?` needed here? I would make sense for S1944, but I think that in this rule, it should work fine without it.
```suggestion
        public void Nullable_In_Arm(string s)
```